docroot/modules/custom/mass_content/mass_content.module

```diff
@@ -819,10 +819,11 @@ function mass_content_query_collection_all_check_sort_value($value = '') {
  */
 function _mass_content_collection_all_only_future_events(QueryPluginBase $query) {
   $collection = mass_content_get_collection_from_current_page();
-  $only_future = $collection->field_show_only_future_events->value;
-
-  if (!$only_future) {
-    return;
+  if ($collection) {
+    $only_future = $collection->field_show_only_future_events->value;
```

Comment:
In the future, please create a method in a bundle class when you need to access a field like field_show_only_future_events. This makes the code more robust. I know this code pre-exists this PR but its important for us to modernize the codebase as we implement features and bug fixes. An example of a bundle class using PR is https://github.com/massgov/openmass/pull/2336

Reply:
Refactored the code as suggested, thanks.